modules/web/src/app/node-data/component.ts

```diff
@@ -317,13 +317,22 @@ export class NodeDataComponent extends BaseFormValidator implements OnInit, OnDe
           : settings.defaultNodeCount;
 
       this.form.get(Controls.Count).setValue(replicas);
-      if (!this.dialogEditMode) {
+      if (this.dialogEditMode) {
+        if (autoUpdatesEnabled && settings.machineDeploymentOptions.autoUpdatesEnforced) {
+          if (!this.form.get(Controls.UpgradeOnBoot).value) {
+            this.form.get(Controls.UpgradeOnBoot).setValue(autoUpdatesEnabled);
```

Comment:
do we want to enforce that for existing MDs as well?

Reply:
That was part of the issue.